File: app/src/main/java/com/fthdgn/books/dependecy/module/PicassoModule.java
package com.fthdgn.books.dependecy.module;

import com.squareup.picasso.Picasso;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import com.fthdgn.books.api.ImageManager;

@Module
public class PicassoModule {

    @Provides
    @Singleton
    ImageManager imageManager(Picasso picasso) {
        return new ImageManager(picasso);
    }
}

File: app/src/main/java/com/fthdgn/books/dependecy/module/PicassoModule.kt
package com.fthdgn.books.dependecy.module

import com.fthdgn.books.api.ImageManager
import com.squareup.picasso.Picasso
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class PicassoModule {

    @Provides
    @Singleton
    internal fun imageManager(picasso: Picasso): ImageManager =
            ImageManager(picasso)
}
